frontend/src/d3/GraphSimulation.ts

```diff
@@ -187,7 +223,7 @@ export default class {
   drawNodes = () => {
     this.nodeSvg
       .selectAll(nodeClassName)
-      .data(this.nodes)
+      .data(this.nodes, (node: any) => node.id)
```

Comment:
Do we have a type we can apply here that isn't `any` or is this restricted by d3?